src/viewProviders/flinkDatabase.ts

```diff
@@ -76,6 +77,39 @@ export class FlinkDatabaseViewProvider extends MultiModeViewProvider<
     ];
   }
 
+  override getChildren(element?: ArtifactOrUdf): any[] {
+    const children = super.getChildren(element);
+
+    // retrive if there are artifacts but no UDFs to show the info message in UDFs mode.
+    if (
+      !element &&
+      this.currentDelegate.mode === FlinkDatabaseViewProviderMode.UDFs &&
+      children.length === 0 &&
+      this.artifactsDelegate.children.length > 0
+    ) {
+      return [{ _isInfo: true, _count: this.artifactsDelegate.children.length }];
+    }
+
+    return children;
+  }
+
+  override getTreeItem(element: any): TreeItem {
+    // default info message when there are artifacts but no UDFs.
+    if (element._isInfo) {
+      const item = new TreeItem(
+        `Found ${element._count} uploaded artifact${element._count > 1 ? "s" : ""}`,
+        TreeItemCollapsibleState.None,
+      );
+      item.description = "Switch to Flink Artifacts to register UDFs";
+      item.command = { command: "confluent.flinkdatabase.setArtifactsViewMode", title: "" };
+      item.iconPath = new ThemeIcon("info");
+      item.tooltip = "";
+      return item;
+    }
+
+    return super.getTreeItem(element);
+  }
```

Comment:
Let's not add empty-state-like tree items here; that's best done in package.json's `viewsWelcome`